jobqueue/scheduler/scheduler.go

```diff
@@ -248,34 +247,25 @@ func New(name string, config interface{}, logger ...log15.Logger) (*Scheduler, e
 		return nil, Error{name, "New", ErrBadScheduler}
 	}
 
-	var l log15.Logger
-	if len(logger) == 1 {
-		l = logger[0].New()
-	} else {
-		l = log15.New()
-		l.SetHandler(log15.DiscardHandler())
-	}
-	s.Logger = l
-
 	s.Name = name
 	s.limiter = make(map[string]int)
-	err := s.impl.initialize(config, l)
+	err := s.impl.initialize(ctx, config)
```

Comment:
For all of the `impl.method(ctx, ...)` calls, we should pass s.typeContext(), where that returns `clog.ContextWithSchedulerType(s.Name)`.